src/libsystemd-network/sd-dhcp-lease.c

```diff
@@ -628,6 +628,8 @@ static int lease_parse_dnr(const uint8_t *option, size_t len, sd_dns_resolver **
                 r = lease_parse_dns_name(option + offset, ilen, &res.auth_name);
                 if (r < 0)
                         return r;
+                if (!dns_name_is_valid_ldh(res.auth_name))
```

Comment:
dns_name_is_valid_ldh() can actually fail for OOM, hence you need to handle negative return codes, i.e. like this:

```c
r = dns_name_is_valid_ldh(res.auth_name);
if (r < 0)
        return r;
if (!r)
        return -EBADMSG;
```

here, and below, and above